src/tests/class_tests/openms/source/PeptideIdentification_test.cpp

```diff
@@ -443,6 +446,52 @@ START_SECTION(static std::vector<PeptideHit> getReferencingHits(const std::vecto
 }
 END_SECTION
 
+START_SECTION((static std::multimap<String, std::pair<Size, Size>> fillConsensusPepIDMap(const ConsensusMap &cmap)))
+{
+  ConsensusMap cmap;
+  ConsensusXMLFile().load(OPENMS_GET_TEST_DATA_PATH("MQEvidence_1.consensusXML"), cmap);
+  std::multimap<String, std::pair<Size, Size>> map_of_UIDs = PeptideIdentification::fillConsensusPepIDMap(cmap);
+
+  auto b = map_of_UIDs.begin();
+  TEST_EQUAL(b->first,
+             "file:///C:/Users/bielow/AppData/Local/Temp/20190911_110348_8204_1/Untitled_workflow/002_FileFilter/out/ES-0014b_2.mzMLspectrum=112")
+  TEST_EQUAL(b->second.first, Size(-1))
+  TEST_EQUAL(b->second.second, 4)
+  ++b;
+  TEST_EQUAL(b->first,
+             "file:///C:/Users/bielow/AppData/Local/Temp/20190911_110348_8204_1/Untitled_workflow/002_FileFilter/out/ES-0014b_2.mzMLspectrum=113")
+  TEST_EQUAL(b->second.first, 11)
+  TEST_EQUAL(b->second.second, 0)
+  ++b;
+  TEST_EQUAL(b->first,
+             "file:///C:/Users/bielow/AppData/Local/Temp/20190911_110348_8204_1/Untitled_workflow/002_FileFilter/out/ES-0014b_2.mzMLspectrum=118")
+  TEST_EQUAL(b->second.first, 4)
+  TEST_EQUAL(b->second.second, 0)
+
+  FeatureMap fmap;
+  FeatureXMLFile().load(OPENMS_GET_TEST_DATA_PATH("MQEvidence_1.featureXML"), fmap);
+  ProteinIdentification::Mapping mp_c(cmap.getProteinIdentifications());
+  const map<String, StringList>& identifier_to_msrunpath = mp_c.identifier_to_msrunpath;
+
+
+  String uid_zero = PeptideIdentification::buildUIDFromPepID(cmap[0].getPeptideIdentifications()[0], identifier_to_msrunpath);
+  String uid_one = PeptideIdentification::buildUIDFromPepID(cmap[0].getPeptideIdentifications()[1], identifier_to_msrunpath);
+  String uid_two = PeptideIdentification::buildUIDFromPepID(cmap[0].getPeptideIdentifications()[2], identifier_to_msrunpath);
+
+  TEST_EQUAL(uid_zero,"file:///C:/Users/bielow/AppData/Local/Temp/20190911_110348_8204_1/Untitled_workflow/002_FileFilter/out/ES-0014b_2.mzMLspectrum=219")
```

Comment:
what I don't like about this UID is that you can't easily go back from the string to path and native ID.
I would suggest to use | as separator as in "file:///C:/test.mzML|spectrum=123"

Reply:
format of the UID should also be documented in the respective class